contracts/escrow-fee-distributor/tests/integration.rs

```diff
@@ -0,0 +1,703 @@
+            resp.events[1].attributes[3].clone(),
+        ]
+    );
+}
+
+#[test]
+fn test_checkpoint_total_supply() {
+    let mut router = mock_app();
+    let router_ref = &mut router;
+
+    let base_pack = init_astroport_test_package(router_ref).unwrap();
+    let escrow_fee_distributor = base_pack.escrow_fee_distributor.unwrap().address;
+    router_ref
+        .execute_contract(
+            Addr::unchecked(USER1.clone()),
+            escrow_fee_distributor.clone(),
+            &ExecuteMsg::CheckpointTotalSupply {},
+            &[],
+        )
+        .unwrap();
+
+    // checks if voting supply per week is set to zero
+    let resp_config: Vec<Uint128> = router_ref
+        .wrap()
+        .query_wasm_smart(&escrow_fee_distributor, &QueryMsg::VotingSupplyPerWeek {})
+        .unwrap();
+
+    let voting_supply: Vec<Uint128> = vec![Uint128::new(0); 1];
+    assert_eq!(voting_supply, resp_config);
+}
+
+#[test]
+fn claim() {
+    let mut router = mock_app();
+    let router_ref = &mut router;
+    let owner = Addr::unchecked(OWNER.clone());
+    let user1 = Addr::unchecked(USER1.clone());
+    let user2 = Addr::unchecked(USER2.clone());
+    let user3 = Addr::unchecked(USER3.clone());
+
+    let base_pack = init_astroport_test_package(router_ref).unwrap();
+
+    // sets 100 ASTRO tokens to distributor (simulate receive astro from maker)
+    BaseAstroportTestPackage::mint(
+        router_ref,
+        owner.clone(),
+        base_pack.astro_token.clone().unwrap().address,
+        &base_pack.escrow_fee_distributor.clone().unwrap().address,
+        100,
+    );
+
+    // checks if distributor's ASTRO token balance is equal to 100
+    BaseAstroportTestPackage::check_balance(
+        router_ref,
+        &base_pack.astro_token.clone().unwrap().address,
+        &base_pack.escrow_fee_distributor.clone().unwrap().address,
+        100,
+    );
+
+    let xastro_token = base_pack.get_staking_xastro(router_ref);
+
+    // sets 200 * 1000_000 xASTRO tokens to user1
+    BaseAstroportTestPackage::mint(
+        router_ref,
+        base_pack.staking.clone().unwrap().address,
+        xastro_token.clone(),
+        &user1,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // checks if user1's xASTRO token balance is equal to 200 * 1000_000
+    BaseAstroportTestPackage::check_balance(
+        router_ref,
+        &xastro_token.clone(),
+        &user1,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // sets 200 * 1000_000 xASTRO tokens to user2
+    BaseAstroportTestPackage::mint(
+        router_ref,
+        base_pack.staking.clone().unwrap().address,
+        xastro_token.clone(),
+        &user2,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // checks if user2's xASTRO token balance is equal to 200 * 1000_000
+    BaseAstroportTestPackage::check_balance(
+        router_ref,
+        &xastro_token.clone(),
+        &user2,
+        (200 * MULTIPLIER) as u128,
+    );
+
+    // locks 100 vxASTRO from user1 for WEEK * 2
```

Comment:
Actually if will be more than 100 vxASTRO but for consistensy just leave comment "// locks 100 xASTRO from user1 for WEEK * 2"